superset-frontend/src/dashboard/components/RefreshIntervalModal.tsx

```diff
@@ -111,17 +157,89 @@ class RefreshIntervalModal extends React.PureComponent<
         modalTitle={t('Refresh interval')}
         modalBody={
           <div>
-            <FormLabel>{t('Refresh frequency')}</FormLabel>
-            <Select
-              ariaLabel={t('Refresh interval')}
-              options={refreshIntervalOptions.map(option => ({
-                value: option[0],
-                label: t(option[1]),
-              }))}
-              value={refreshFrequency}
-              onChange={this.handleFrequencyChange}
-              sortComparator={propertyComparator('value')}
-            />
+            <div id="refresh_from_dropdown">
+              <FormLabel>
+                <b>{t('Refresh frequency')}</b>
+              </FormLabel>
+              <Select
+                ariaLabel={t('Refresh interval')}
+                options={this.createIntervalOptions(refreshIntervalOptions)}
+                value={refreshFrequency}
+                onChange={this.handleFrequencyChange}
+                sortComparator={propertyComparator('value')}
+              />
+            </div>
+            <div
+              style={{
+                visibility: refreshFrequency === -1 ? 'visible' : 'hidden',
+                display: 'flex',
+                gap: '3%',
+                marginTop: '15px',
+              }}
+              id="custom_block_view"
```

Comment:
Hi, instead of using css to hide/show can we use React to conditionally render the element?

Reply:
I am changing views based on dropdown selected options, i was trying by using react but it was not working,
I am not sure about this, but i will check and try to do if possible.

Reply:
What is that is not working specifically? We definitely need to show this conditionally and not hidden via CSS